build/terraform/e2e/module.tf

```diff
@@ -28,6 +28,10 @@ terraform {
       version = "~> 2.3"
     }
   }
+  backend "gcs" {
+   bucket  = "agones-e2e-infra-bucket-tfstate"
```

Comment:
Given that a bucket name is unique to the project - this should probably be dynamic (use the project name as a prefix?) so it can be run against development projects without failing.

Reply:
Yeah, that's doable. But I have doubt whether it makes sense to create an e2e test cluster for development. I was thinking we should push user to just create one test-cluster and use it for the development purpose even the e2e test.

Reply:
With it the way it is, you can't test this Terraform script, except to test it in production.

Reply:
Yeah, I was expecting the backend part would be relatively stable since it's only a thing for our maintainers who have access to the infra and won't have many change (if any). But it's still a fair point. I will make it not hardcoded.

Reply:
Honestly just something like `{project}-tfstate` makes sense to me, that's what I was thinking of doing.